slack_sdk/socket_mode/async_client.py

```diff
@@ -116,7 +127,7 @@ async def run_message_listeners(self, message: dict, raw_message: str) -> None:
             )
         try:
             if message.get("type") == "disconnect":
-                await self.connect_to_new_endpoint()
+                await self.connect_to_new_endpoint(force=True)
```

Comment:
The sync client does the same. When we receive a disconnect type message from the Socket Mode server-side, we should immediately reconnect even if the current session is still active.